xcode/Subconscious/Shared/Services/TranscludeService.swift

```diff
@@ -0,0 +1,68 @@
+//
+//  TranscludeService.swift
+//  Subconscious
+//
+//  Created by Ben Follington on 5/5/2023.
+//
+
+import Foundation
+import Combine
+
+actor TranscludeService {
+    private var database: DatabaseService
+    private var noosphere: NoosphereService
+    
+    init(database: DatabaseService, noosphere: NoosphereService) {
+        self.database = database
+        self.noosphere = noosphere
+    }
+    
+    func fetchTranscludePreviews(
+        slashlinks: [Slashlink],
+        owner: UserProfile
+    ) async throws -> [Slashlink: EntryStub] {
+        let petname = owner.address.petname
+        
+        let slashlinks = slashlinks.map { address in
+            guard case .petname(_) = address.peer else {
+                // Rebase relative slashlinks to the owner's handle (if they have one)
+                return Slashlink(peer: owner.address.peer, slug: address.slug)
+            }
+            
+            return address
+        }
+        
+        let entries = try database.listEntries(for: slashlinks, owner: petname)
+        
+        return
+            Dictionary(
+                entries.map { entry in
```

Comment:
Nit: it might be marginally more efficient to create a mutable dict and set keys one-by-one in a for loop. Avoids creating interim array and tuples. However this is a micro-optimization and I'm not sure it is worthwhile. If worse for readability, forget it.